### ISSUE TITLE
[Optimizations] Try to reduce peak memory usage

### ISSUE THREAD

**Issue by alloy** [artsy/metaphysics#489]
Besides our throughput going down when under pressure, we’d also like to run on smaller Heroku dynos.

@craigspaeth Can you paste dump any of your current investigations here?

**Comment by craigspaeth** [comment]
It appears we have an occasional run-away memory leak not obviously associated with spikes in throughput.

![image](https://cloud.githubusercontent.com/assets/555859/20951389/77c83390-bbf3-11e6-84f1-158011f61a72.png)

I do see a correlation in Gravity response time with the most recent memory spike, but as you can see below it doesn't correspond with the prior two spikes:

![image](https://cloud.githubusercontent.com/assets/555859/20951737/95f1bec0-bbf5-11e6-8bf1-918de2e33daa.png)
 
Because NewRelic doesn't breakdown by GraphQL query, it's tricky to get much insight from it. We could try implementing [Apollo Optics](http://www.apollodata.com/optics) to do better, but we might also be able to implement some manual reporting to NR to get enough of a breakdown.

These spikes only started after the last deploy which comes with these changes...

![image](https://cloud.githubusercontent.com/assets/555859/20951881/923b05ba-bbf6-11e6-9022-06aa0d12e1dd.png)

...digging through those PRs, I'm pretty convinced it's in [this PR](https://github.com/artsy/metaphysics/pull/479) with one of the updated deps.

**Comment by alloy** [comment]
> Because NewRelic doesn't breakdown by GraphQL query, it's tricky to get much insight from it. We could try implementing Apollo Optics to do better, but we might also be able to implement some manual reporting to NR to get enough of a breakdown.

I agree that we need this first, we need to be able to reproduce the leak to diagnose it.

I’ll get on that first.
